pdr_utils/subgraph.py

```diff
@@ -116,3 +115,48 @@ def get_all_interesting_prediction_contracts():
             print(e)
             return {}
     return contracts
+
+
+def get_consume_so_far(predictoor_contracts, week_start_timestamp, consumer_address):
+    chunk_size = 1000  # max for subgraph = 1000
+    offset = 0
+    consume_so_far = 0
+    while True:
+        query = """
+        {
+            predictContracts(skip:%s, first:%s){
+                id	
+                token{
+                    orders(where: {createdTimestamp_gt:%s, consumer_in:["%s"]}){
+        		        createdTimestamp
+                        consumer {
+                            id
+                        }
+                        lastPriceValue
+                    }
+                }
+            }
+        }
+        """ % (
+            offset,
+            chunk_size,
+            week_start_timestamp,
+            consumer_address.lower(),
+        )
+        offset += chunk_size
+        try:
+            result = query_subgraph(query)
+            new_orders = result["data"]["predictContracts"]
+            if new_orders == []:
+                break
+            for order in new_orders:
+                if order["id"] in predictoor_contracts:
+                    if len(order["token"]["orders"]) > 0:
```

Comment:
can we do the indentation trick with the ifs here? i.e. if order["id"] not in predictoor contracts: continue, same for if len == 0... then we can unindent the inner for